Zeroconf renew.

kridency opened the pull request:
Due to some API changes in rust zeroconf module there should be done some refactoring in mdns.rs

Commit:
Zeroconf renew.

kridency committed:
Merge branch 'master' into master

jkcoxson commented on librusb.rlib:
We shouldn't be adding binary files to the repository, not sure how this got here

biofreak replied:
Deleted.
commit 57b380b98e4b8552bf96a655921ad88fbd9a2268

jkcoxson commented on netmuxd:
We shouldn't be adding binary files to the repository, not sure how this got here

biofreak replied:
Deleted.
commit 57b380b98e4b8552bf96a655921ad88fbd9a2268

jkcoxson commented on Cargo.toml:
```diff
@@ -25,11 +25,9 @@ log = { version = "0.4.16" }
-usb = ["libusb1-sys", "rusb"]
-vendored = ["rusty_libimobiledevice/vendored", "plist_plus/vendored"]
+usb = [ "libusb1-sys", "rusb" ]
+default = ["rusb", "zeroconf" ]
```
We shouldn't be adding rusb to the default features, it's not even implemented

biofreak replied:
rusb replaced for libusb1-sys in default.
commit 57b380b98e4b8552bf96a655921ad88fbd9a2268

jkcoxson commented on Cargo.toml:
```diff
@@ -15,8 +15,8 @@ path = "src/passthrough.rs"
-zeroconf = { git = "https://github.com/zeyugao/zeroconf-rs", optional = true }
-mdns = "3.0.0"
+zeroconf = { version = "0.14.2", optional = true }
+mdns = { version = "3.0.0", optional = true }
```
Does the crate still work with the mdns dependency?

biofreak replied:
Removed mdns crate.
commit 57b380b98e4b8552bf96a655921ad88fbd9a2268

jkcoxson commented on src/mdns.rs:
```diff
@@ -3,64 +3,89 @@
+}
+
+#[cfg(feature = "zeroconf")]
+fn on_service_discovered(result: zeroconf::Result<ServiceDiscovery>, _context: Option<Arc<dyn Any>>,) {
```
Context shouldn't start with a _ as it's used

biofreak replied:
Renamed variables.
commit 57b380b98e4b8552bf96a655921ad88fbd9a2268

jkcoxson commented on src/mdns.rs:
```diff
@@ -3,64 +3,89 @@
 use std::net::IpAddr;
-use std::sync::Arc;
+use std::any::Any;
+use std::sync::{Arc, Mutex as StdMut};
```
I would prefer not use import alias's (StdMut)

biofreak replied:
Removed alias.
commit 57b380b98e4b8552bf96a655921ad88fbd9a2268

Commit:
Changes based on code review.

jkcoxson committed:
Restore features to Cargo.toml

jkcoxson committed:
mdns.rs: Use String::new instead of from slice

jkcoxson commented on Cargo.toml:
```diff
@@ -15,8 +15,8 @@ path = "src/passthrough.rs"
 
-zeroconf = { git = "https://github.com/zeyugao/zeroconf-rs", optional = true }
-mdns = "3.0.0"
+zeroconf = { version = "0.14.2", optional = true }
```
It looks like the newest version of zeroconf is 0.15. Is there a reason we're not targeting the latest version?

biofreak replied:
Not at all.

Commit:
Refresh zeroconf crate version.

Commit:
Merge branch 'master' of ssh://github.com/biofreak/netmuxd

jkcoxson committed:
Restore Zeroconf as an optional dependency Cargo.toml

Zeroconf is only supported on Linux, so it's an optional feature. I forgot that.

jkcoxson commented:
If someone else can confirm this patch works, it looks reasonable to me. I don't have an iDevice to test it with right now.

jkcoxson commented:
Zeroconf has been removed in favor of dns-sd
